app/state_machines/form_state_machine.rb

```diff
@@ -40,13 +42,12 @@ module FormStateMachine
       end
 
       event :archive_live_form do
-        # TODO: Temporary transition until we do archive state properly so that live forms cant
-        # be accessed but still exist in the database
-        # https://trello.com/c/QXtBIrKc/1342-add-archived-states-to-forms-state-machine
-        transitions from: %i[live live_with_draft], to: :draft
+        after do
+          made_live_forms.destroy_all
+        end
```

Comment:
we dont want to delete live forms, the form state will control `live_version` method so it either serves the live version or it doesn't if the forms state is archived

Reply:
Cool, changed it to achieve this